File: tests/sample_ts_repo/src/repository.ts
import { Order } from "./models";

export class OrderRepository {
  private orders = new Map<string, Order>();

  save(order: Order): void {
    this.orders.set(order.idempotencyKey, order);
  }

  findByKey(idempotencyKey: string): Order | undefined {
    return this.orders.get(idempotencyKey);
  }
}

File: tests/sample_go_repo/orders/repository.go
package orders

type OrderRepository struct {
	orders map[string]*Order
}

func NewRepository() *OrderRepository {
	return &OrderRepository{orders: map[string]*Order{}}
}

func (r *OrderRepository) Save(order *Order) {
	r.orders[order.IdempotencyKey] = order
}

func (r *OrderRepository) FindByKey(idempotencyKey string) *Order {
	return r.orders[idempotencyKey]
}
